documentation/create_overview_doc/createdoc.sh

```diff
@@ -111,419 +82,215 @@ function read_skip_list() {
-            while [ $attempt -le $MAX_RETRIES ]; do
-                echo "Attempt $attempt: Processing Mermaid script for $mdd_file" >&2
-                # Use bito to process the Mermaid script
-                temp_file=$(mktemp)
-                echo -e "$mermaid_script" | bito -p "$mermaid_doc_prompt_file" > "$temp_file"
-
-                # Validate the Mermaid script
-                if validate_mermaid_syntax "$(cat "$temp_file")"; then
-                    echo -e "Valid Mermaid diagram generated successfully for $mdd_file.\n" >&2
-                    # Update the latest valid Mermaid content
-                    latest_valid_mermaid_content=$(cat "$temp_file")
-                    rm "$temp_file"
-                    # Delete the processed .mdd file
-                    rm "$mdd_file"
-                    update_token_usage "$mermaid_script" "$latest_valid_mermaid_content"
-                    break
-                else
-                    echo -e "Invalid Mermaid diagram syntax for attempt $attempt. Retrying...\n" >&2
-                    rm "$temp_file"
-                    sleep $RETRY_DELAY
-                    ((attempt++))
-                fi
-            done
-
-            if [ $attempt -gt $MAX_RETRIES ]; then
-                echo "Failed to generate a valid Mermaid diagram for $mdd_file after $MAX_RETRIES attempts."
-                return 1
-            fi
-        fi
-    done
+    # Generate a flowchart for the entire system
+    echo "Generating system flowchart..."
+    code2flow "$source_folder" -o "$output_file" --hide-legend --no-trimming --verbose --skip-parse-errors
 
-    # Check if there is valid Mermaid content
-    if [ -n "$latest_valid_mermaid_content" ]; then
-        # Save the latest valid Mermaid content to overview.mdd
-        echo -e "$latest_valid_mermaid_content" > "$overview_mdd_file"
-        echo "Mermaid overview generated successfully: $overview_mdd_file"
+    # Check if the flowchart was successfully generated
+    if [ -f "$output_file" ]; then
+        echo "System flowchart created successfully"
     else
-        echo "Failed to create overview.mdd or no valid diagrams were found."
+        echo "Failed to create system flowchart" >&2
         return 1
     fi
 }
 
-# Function to read extensions from CSV and create the find command
-function create_find_command() {
-    local lang_file="$1"
-    local folder_to_document="$2"
-    local find_command="find \"$folder_to_document\" -type f"
-
-    # Read each line from CSV and append it to the find command
-    while IFS= read -r ext; do
-        find_command="$find_command -o -name \"*.$ext\""
-    done < "$lang_file"
-
-    # Correct the find command by adding parentheses and removing the first '-o'
-    find_command="${find_command/-o /\\( } \)"
-    echo "$find_command"
-}
-
 # Capture Start Time
 start_time=$(date +%s)
 
 function main() {
+    echo "Verifying prerequisites before documentation generation..." >&2
```

Comment:
**Suggestion**: Before starting the documentation generation process, it's good practice to verify not only the presence of required tools and files but also their versions if specific features or capabilities are needed. Consider adding version checks for critical dependencies like Code2Flow and Graphviz. <br> **Code Suggestion**: 
 ```
 code2flow_version=$(code2flow --version)
 if [[ $code2flow_version < 'Expected_Min_Version' ]]; then
     echo "Error: code2flow version must be at least 'Expected_Min_Version'. Current version: $code2flow_version."
     exit 1
 fi
 ```